### ISSUE TITLE
Support multiple vouchers and/or gift cards during checkout

### ISSUE THREAD

**Issue by khalibloo** [saleor/saleor#14995]
## Problem

During checkout, a user may want to apply multiple vouchers, or a voucher and a gift card for example. However, at least from the checkout query, there is only a single `voucherCode` field. This leads me to believe only 1 voucher code can be added to a checkout. In my case, I need to apply and query multiple voucher codes in a checkout.

---
## 2. API changes
Checkouts should have an array of voucher codes instead of a single string.

### Types

```graphql
{
  checkout {
    promoCodes // string[]
    // or
    voucherCodes // string[]
    giftCardCodes // string[]
  }
}
```

**Comment by SarthakNikhal** [comment]
@patrys Does this need to be done? If so, I'd like to work on it.

**Comment by DustinBracy** [comment]
+1, I would love to see this feature implemented.

**Comment by MikhailKazan** [comment]
@khalibloo do you find the solution for multiple voucher codes on checkout?

**Comment by khalibloo** [comment]
@MikhailKazan no, I couldn't find a way around this limitation unfortunately.

**Comment by MikhailKazan** [comment]
@khalibloo I think about the possibility to use voucher_code string to utilize space separated single codes. Then the parsing and calculation should be changed.
